invenio_rdm_records/resources/serializers/schemaorg/schema.py

```diff
@@ -0,0 +1,343 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+# Copyright (C) 2021 Northwestern University.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Schemaorg based Schema for Invenio RDM Records."""
+from typing import Optional
+
+import pycountry
+from commonmeta import dict_to_spdx, doi_as_url, parse_attributes, unwrap, wrap
+from edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+from flask_resources.serializers import BaseSerializerSchema
+from idutils import to_url
+from invenio_access.permissions import system_identity
+from invenio_records_resources.proxies import current_service_registry
+from marshmallow import Schema, ValidationError, fields, missing
+from marshmallow_utils.fields import SanitizedUnicode
+from pydash import py_
+
+from ..schemas import CommonFieldsMixin
+from ..utils import convert_size, get_vocabulary_props
+
+
+class PersonOrOrgSchema(Schema):
+    """Creator/contributor schema for Schemaorg."""
+
+    name = fields.Str(attribute="person_or_org.name")
+    givenName = fields.Str(attribute="person_or_org.given_name")
+    familyName = fields.Str(attribute="person_or_org.family_name")
+    affiliation = fields.Method("get_affiliation")
+    id_ = fields.Method("get_name_identifier", data_key="@id")
+    type_ = fields.Method("get_name_type", data_key="@type")
+
+    #
+    # Private helper functions
+    #
+
+    def _serialize_identifier(self, identifier):
+        """Format on name identifier."""
+        if not identifier.get("identifier"):
+            return None
+
+        if identifier.get("scheme") == "isni":
+            return "http://isni.org/isni/" + identifier.get("identifier")
+
+        # Schemaorg expects a URL for the identifier
+        return to_url(identifier["identifier"], identifier["scheme"], "https")
+
+    #
+    # Schema Methods
+    #
+
+    def get_name_type(self, obj):
+        """Get name type. Schemaorg expects either Person or Organisation.
+
+        Defaults to Person.
+        """
+        name_type = py_.get(obj, "person_or_org.type")
```

Comment:
```suggestion
        from invenio_records.dictutils import dict_lookup
        name_type = dict_lookup(obj, "person_or_org.type")
```

minor: to avoid adding the `pydash` dependency.

Reply:
`dict_lookup` raises if it doesn't exist. I fear we would have a lot of:

```python

try:
    x = dict_lookup(obj, "a.b.c.")
except KeyError:
    pass
```

It would be nice if we could pass a `default` and only raise if there is no default.